### PR TITLE
rdf/sliceddensity analyses, electrostatic self-energy, fene vs fene+wca, bondenergy in speciation

### PR DESCRIPTION
**Pull request by gitesei**
- option to calculate atomic RDF in a slice
- sliced density w.r.t. COM of a given atom type
- self energy of Coulomb Galore potentials adapted to GC simulations
- ignore bonded energy in Metropolis criterion for Speciation moves
- L/R reaction coordinate for bending modulus calculations
- distinct FENE and FENE+WCA bonded potentials (the latter is useful in combination with nonbonded_cached)

### CODE REVIEW

File: src/analysis.h [hunk 1]
```diff
@@ -977,6 +976,7 @@ namespace Faunus {
                         from_json(j);
                         name = "savestate";
                         steps = j.value("nstep", -1);
+                        saverandom = j.value("saverandom", false);
```
Comment (mlund) [inline src/analysis.h]: add to documentation?

File: src/analysis.cpp [hunk 2]
```diff
@@ -153,21 +153,25 @@ void Faunus::Analysis::PairFunctionBase::_to_json(Faunus::json &j) const {
...
     assertKeys(j, {
-            "file", "name1", "name2", "dim", "dr", "Rhyper", "nstep", "nskip"
+            "file", "name1", "name2", "dim", "dr", "Rhyper", "nstep", "nskip", "slicedir", "thickness"
             });
     file = j.at("file");
     name1 = j.at("name1");
     name2 = j.at("name2");
     dim = j.value("dim", 3);
     dr = j.value("dr", 0.1) * 1.0_angstrom;
+    slicedir = j.value("slicedir", slicedir);
```
Comment (mlund) [inline src/analysis.cpp]: add to documentation?

File: src/analysis.h [hunk 3]
```diff
@@ -530,39 +530,37 @@ namespace Faunus {
...
+                std::string atomCOM;
+                int idCOM = -1; // center at COM of idCOM atoms?
 
                 void _from_json(const json &j) override {
                     file = j.at("file").get<std::string>();
                     names = j.at("atoms").get<decltype(names)>(); // molecule names
                     ids = names2ids(atoms, names);     // names --> molids
                     dz = j.value("dz", 0.1);
-                    COM = j.value("com", false);
+                    if (j.count("atomcom") == 1) {
```
Comment (mlund) [inline src/analysis.h]: add to documentation?

File: src/penalty.h [hunk 4]
```diff
@@ -98,10 +97,19 @@ namespace Faunus {
...
                         if (property=="x") f = [&p=spc.p, i=index]() { return p[i].pos.x(); };
-                        if (property=="y") f = [&p=spc.p, i=index]() { return p[i].pos.y(); };
-                        if (property=="z") f = [&p=spc.p, i=index]() { return p[i].pos.z(); };
-                        if (property=="R") f = [&p=spc.p, i=index]() { return p[i].pos.norm(); };
-                        if (property=="q") f = [&p=spc.p, i=index]() { return p[i].charge; };
+                        else if (property=="y") f = [&p=spc.p, i=index]() { return p[i].pos.y(); };
+                        else if (property=="z") f = [&p=spc.p, i=index]() { return p[i].pos.z(); };
+                        else if (property=="R") f = [&p=spc.p, i=index]() { return p[i].pos.norm(); };
+                        else if (property=="q") f = [&p=spc.p, i=index]() { return p[i].charge; };
+                        else if (property=="N") // number of atom of id=index
```
Comment (mlund) [inline src/penalty.h]: document

File: src/penalty.h [hunk 5]
```diff
@@ -191,6 +200,30 @@ namespace Faunus {
                             };
                         }
 
+                        else if (property=="L/R") {
```
Comment (mlund) [inline src/penalty.h]: document

Comment (gitesei) [reply src/penalty.h]: Thanks, Mikael! The docs are now updated.

**Pull request merged**